impeller/display_list/canvas.cc

```diff
@@ -648,34 +649,84 @@ void Canvas::DrawCircle(const Point& center,
   }
 }
 
-void Canvas::ClipGeometry(std::unique_ptr<Geometry> geometry,
+void Canvas::ClipGeometry(const Geometry& geometry,
```

Comment:
I combined AddClipEntityToCurrentPass into this method, since its all only called in one place.